backends/ent/backend.go

```diff
@@ -0,0 +1,90 @@
+// --------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 The Protobom Authors
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// --------------------------------------------------------------
+package ent
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"slices"
+
+	sqlite "github.com/glebarez/go-sqlite"
+	"github.com/protobom/protobom/pkg/storage"
+
+	"github.com/protobom/storage/internal/backends/ent"
+)
+
+// Backend implements the protobom.pkg.storage.Backend interface.
+type Backend struct {
+	// BackendOptions is the set of options common to all ent Backends.
+	*BackendOptions
+}
+
+var _ storage.Backend = (*Backend)(nil)
+
+func NewBackend(opts ...Option) *Backend {
+	backend := &Backend{
+		BackendOptions: NewBackendOptions(),
+	}
+
+	for _, opt := range opts {
+		opt(backend)
+	}
+
+	backend.init(backend.BackendOptions)
+
+	return backend
+}
+
+func (backend *Backend) init(opts *BackendOptions) {
+	if backend.BackendOptions == nil {
+		backend.BackendOptions = NewBackendOptions()
+	}
+
+	// Register the SQLite driver as "sqlite3".
+	if !slices.Contains(sql.Drivers(), "sqlite3") {
+		sqlite.RegisterAsSQLITE3()
+	}
+
+	clientOpts := []ent.Option{}
+	if opts.debug {
+		clientOpts = append(clientOpts, ent.Debug())
+	}
+
+	client, err := ent.Open("sqlite3", opts.DatabaseFile+dsnParams, clientOpts...)
+	if err != nil {
+		panic(fmt.Errorf("failed opening connection to sqlite: %w", err))
```

Comment:
Don't panic here. This needs to return an error and let the application handle it. 
```suggestion
		return fmt.Errorf("failed opening connection to sqlite: %w", err)
```